app/views/shared/_user_metadata.html.erb

```diff
@@ -1,83 +1,83 @@
 <div class="user-metadata-details">
```

Comment:
I'm not super convinced this refactoring is correct. That's because half of this fail checks for `@organization` and the other half checks for `@user`. These variables unfortunately used to come from here: 

https://github.com/forem/forem/blob/f8e56b356eb66b70b5ba96f63a3f4d614c29fee0/app/controllers/stories_controller.rb#L237

https://github.com/forem/forem/blob/f8e56b356eb66b70b5ba96f63a3f4d614c29fee0/app/controllers/stories_controller.rb#L263-L264

and they both could be present, so now that you only pass 1 object as `actor`, it could fail, because org and user have different fields. For example, if `actor` is an org, `organization.email_public` doesn't exist, hence it will raise an exception.

Reply:
I think it could maybe use some more refactoring to make it clear, but `user_metadata` is only rendered in two places:

- `profile_preview_card_content`
- `sticky_nav` (the right-hand sidebar on the article page)

The `sticky_nav` sets a variable called actor via `<% @actor = @article.organization || @article.user %>` and passed it to the metadata. For profile preview cards, we take the actor from either the endpoint added in this PR / get it passed directly as the user

So I think it is working 🤔  I checked the UI for an article made by an organisation and it looked OK

Reply:
That's weeird.

Reply:
I can take another look at it 😄  

I think the problem I will have is that for an article which is written by e.g. Admin McAdmin for Organisation X, the previous implementation would have assumed any preview card would be for Organisation X. But actually I want one for Admin McAdmin

Reply:
Going to check with product about this code generally - looking at it further, I notice it's never being used in an organization context, so maybe I can simplify this a bunch

Reply:
@aitchiss, to get this merged, are we waiting on product, or would that extra bit come in a subsequent PR?

Reply:
I think as long as we're happy that this PR is behaving as expected for:
- organization cards in the right-hand sticky nav, 
- authors of a post published under an organization,
- comment authors 

then we can address the metadata refactor in a subsequent PR rather than waiting on product. Especially since this PR doesn't close off the RFC completely.

The current refactor will at least fix the existing bug in production where some organization metadata is being shown on the author byline preview card (if they published under an organization)